app/controllers/course/stage/instructions.ts

```diff
@@ -54,13 +54,17 @@ export default class CourseStageInstructionsController extends Controller {
   }
 
   get shouldShowLanguageGuide() {
-    return !this.model.courseStage.isFirst && !!this.languageGuide && this.featureFlags.canSeeLanguageGuidesForStage2;
+    return !this.model.courseStage.isFirst && !!this.languageGuide;
   }
 
   get shouldShowPrerequisites() {
     return !!this.prerequisiteInstructionsMarkdown;
   }
 
+  get shouldShowSolution() {
+    return this.featureFlags.canSeeSolutionsForStage2;
```

Comment:
```suggestion
    return this.featureFlags.canSeeSolutionsForStage2 && this.repository.secondStageSolution;
```

(Consistent with what we're doing in the other getters, and ensures we don't accidentally rely on the property in some place where a solution isn't available)